### PR TITLE
2070159: Update resource locator annotation warnings

### PR DESCRIPTION
**Pull request by wottop**
Checks the interfaces for the Resource classes
Downgraded log warning to info for PUT/POST with no @Consumes as this
 is the expected state

### CODE REVIEW

**Comment by JoshAlbrecht23** [comment]
These changes look good to me, but I'd like to get a second opinion!

File: src/main/java/org/candlepin/resteasy/ResourceLocatorMap.java [hunk 1]
```diff
@@ -234,7 +240,8 @@ protected void logLocators() {
                 continue;
             }
 
-            if (!m.isAnnotationPresent(Consumes.class)) {
+            if (!m.isAnnotationPresent(Consumes.class) &&
```
Comment (Januson) [inline src/main/java/org/candlepin/resteasy/ResourceLocatorMap.java:243]: Tests are hitting NPE at line 255. I guess we have to seach for the interface annotation as well.

File: src/main/java/org/candlepin/resteasy/ResourceLocatorMap.java [hunk 2]
```diff
@@ -215,14 +215,20 @@ protected void logLocators() {
             String name = m.getDeclaringClass() + "." + m.getName();
             registered.append("\t" + name + "\n");
 
-            if (!m.isAnnotationPresent(Produces.class)) {
+            if (!m.isAnnotationPresent(Produces.class) &&
```
Comment (Januson) [inline src/main/java/org/candlepin/resteasy/ResourceLocatorMap.java:218]: Could you add some tests? We don't have any for this class.

**Review by Januson — CHANGES_REQUESTED** [review]
Changes look good, but it looks like they introduced a NPE.

**Review by Januson — APPROVED** [review]

**Pull request merged**